### PR TITLE
Make OneOf structs readonly

### PR DESCRIPTION
**Pull request by XanderStoffels**
Solves #120, improving performance when passing a `struct` to a method using the `in` keyword. 

[MS Docs](https://learn.microsoft.com/en-us/dotnet/csharp/write-safe-efficient-code#declare-immutable-structs-as-readonly)
[Blogpost with more insight](https://devblogs.microsoft.com/premier-developer/avoiding-struct-and-readonly-reference-performance-pitfalls-with-errorprone-net/)
[Stack overflow discussion](https://stackoverflow.com/questions/50539764/dodging-the-performance-hit-from-using-in-with-a-struct-without-making-the-str)

#### TL;DR
Using `in` for a parameter lets us pass by reference, but then prevents us from assigning a value to it. However the performance can actually become worse, because it creates a "defensive copy" of the `struct`, copying the whole thing. A way around this is to use read-only for a `struct`. When you pass it into an `in` parameter, the compiler sees that it's read-only and won't create the defensive copy, thereby making it the better alternative for performance.

### CODE REVIEW

**Comment by XanderStoffels** [comment]
On a side note, test ` LeftSideFormatsWithCurrentCulture("en-NZ")` and `RightSideFormatsWithCurrentCulture("en-NZ")` seem to be failing for me on the master branch before I introduced changes, so I left those alone. All other tests pass.

**Review by wdolek — COMMENTED** [review]
If I was in charge, I would **approve** and release, this is simple yet powerful change.
